dateutil/test/test_parser.py

```diff
@@ -675,6 +675,41 @@ def test_era_trailing_year(self):
         assert res.year == 2001, res
 
 
+class TestOutOfBounds(object):
+
+    def test_no_year_zero(self):
+        with pytest.raises(ValueError):
+            parse("0000 Jun 20")
+
+    def test_out_of_bound_day(self):
+        with pytest.raises(ValueError):
+            parse("Feb 30, 2007")
+
+    @pytest.mark.parametrize('fuzzy', [True, False])
```

Comment:
I think this is probably fine, but a parametrized fixture may be able to do this with less repetition? I am by no means an expert, but one option is:

```python
@pytest.fixture(scope="class", params=[True, False])
def fuzzy(self, arg):
    return arg.param
```

Then you can just have "fuzzy" be an argument in all the other tests.

Reply:
Sure.  Easier to just put it at the module level and not need the `scope="class"` or `self`